datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -875,23 +875,12 @@ async fn collect_left_input(
 
     // Estimation of memory size, required for hashtable, prior to allocation.
     // Final result can be verified using `RawTable.allocation_info()`
-    //
-    // For majority of cases hashbrown overestimates buckets qty to keep ~1/8 of them empty.
-    // This formula leads to overallocation for small tables (< 8 elements) but fine overall.
-    let estimated_buckets = (num_rows.checked_mul(8).ok_or_else(|| {
```

Comment:
Another alternative is to make `estimate_memory_size` return `Result<usize>` rather than `usize` 

In general if the cost model overflows that is probably something that would be good to flag as an error rather than continuing calculation

Reply:
Yes, I think this would be the cleanest solution. I wrote a comment in the issue, since this would require us to change the Accumulator trait; and I'm not sure we want to do this, for a probably rare edge-case?

Reply:
Perhaps we can simply ignore the error in the accumulator (e.g. do the unwrap in at the callsite) and discuss changing the accumulator trait in some future PR?

Reply:
Sorry -- i missed the [comment](https://github.com/apache/datafusion/issues/8764#issuecomment-2143467627) on the ticket

I think we should change the code to panic on overflow and we can sort out what to do if that is seen in practice

Reply:
so you mean unwrap() the error at callsite (the accumulator)?